main/src/ca/uwaterloo/flix/language/ast/Ast.scala

```diff
@@ -79,6 +79,38 @@ object Ast {
     override def hashCode(): Int = input.hashCode()
   }
 
+  /**
+    * A common supertype for constant values.
+    */
+  sealed trait Constant
+  object Constant {
+    case object Unit extends Constant
+
+    case object Null extends Constant
```

Comment:
I assume the source locations are associated with the Expression?

Reply:
Yes.